main/lsp/LSPPreprocessor.h

```diff
@@ -22,6 +22,10 @@ struct QueueState {
     bool terminate = false;
     bool paused = false;
     int errorCode = 0;
+    // The latest edit version to be processed on the processing thread.
+    u4 latestVersion = 0;
+    // If 'true', the processing thread is currently running a slow path.
+    bool runningSlowPath = false;
```

Comment:
let's make this `std::shared_ptr<std::atomic<u4>> lastCommitedState;`
and do comparison on it. This will prevent the race